### ISSUE TITLE
Extra singe quotes cause kubectl failure 

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher-cleanup#74]
The following error keeps showing up in the output of the cleanup script:
```
++ kubectl get -o 'jsonpath={.metadata.finalizers}' '-n cattle-fleet-clusters-system Role.rbac.authorization.k8s.io import-token-local-68cdad43-b01a-427f-959d-d0cf2d5797b0-creds'
You must specify the type of resource to get. Use "kubectl api-resources" for a complete list of supported resources.

error: Required resource not specified.
Use "kubectl explain <resource>" for a detailed description of that resource (e.g. kubectl explain pods).
See 'kubectl get -h' for help and examples
```

It looks like the error is caused by the extra single quote around the `-n cattle-fleet-clusters-system ...` part. 

Notes:
I run the cleanup script directly in the command line, instead of deploying the job. I am not sure if that could contribute to the error. 

<details>
<summary>more logs</summary>

```
+ break
+ read -r NAME NAMESPACE KIND APIVERSION
++ printapiversion rbac.authorization.k8s.io/v1
++ echo rbac.authorization.k8s.io/v1
++ grep -q /
++ echo rbac.authorization.k8s.io/v1
++ cut -d/ -f1
+ kcpf -n cattle-fleet-clusters-system Role.rbac.authorization.k8s.io import-token-local-68cdad43-b01a-427f-959d-d0cf2d5797b0-creds
++ kubectl get -o 'jsonpath={.metadata.finalizers}' -n cattle-fleet-clusters-system Role.rbac.authorization.k8s.io import-token-local-68cdad43-b01a-427f-959d-d0cf2d5797b0-creds
+ FINALIZERS='["wrangler.cattle.io/auth-prov-v2-role"]'
+ '[' 'x["wrangler.cattle.io/auth-prov-v2-role"]' '!=' x ']'
+ echo 'Finalizers before for -n cattle-fleet-clusters-system Role.rbac.authorization.k8s.io import-token-local-68cdad43-b01a-427f-959d-d0cf2d5797b0-creds: ["wrangler.cattle.io/auth-prov-v2-role"]'
Finalizers before for -n cattle-fleet-clusters-system Role.rbac.authorization.k8s.io import-token-local-68cdad43-b01a-427f-959d-d0cf2d5797b0-creds: ["wrangler.cattle.io/auth-prov-v2-role"]
+ kubectl patch -p '{"metadata":{"finalizers":null}}' --type=merge -n cattle-fleet-clusters-system Role.rbac.authorization.k8s.io import-token-local-68cdad43-b01a-427f-959d-d0cf2d5797b0-creds
role.rbac.authorization.k8s.io/import-token-local-68cdad43-b01a-427f-959d-d0cf2d5797b0-creds patched
++ kubectl get -o 'jsonpath={.metadata.finalizers}' '-n cattle-fleet-clusters-system Role.rbac.authorization.k8s.io import-token-local-68cdad43-b01a-427f-959d-d0cf2d5797b0-creds'
You must specify the type of resource to get. Use "kubectl api-resources" for a complete list of supported resources.

error: Required resource not specified.
Use "kubectl explain <resource>" for a detailed description of that resource (e.g. kubectl explain pods).
See 'kubectl get -h' for help and examples
+ echo 'Finalizers after for -n cattle-fleet-clusters-system Role.rbac.authorization.k8s.io import-token-local-68cdad43-b01a-427f-959d-d0cf2d5797b0-creds: '
Finalizers after for -n cattle-fleet-clusters-system Role.rbac.authorization.k8s.io import-token-local-68cdad43-b01a-427f-959d-d0cf2d5797b0-creds:
++ printapiversion rbac.authorization.k8s.io/v1
++ echo rbac.authorization.k8s.io/v1
++ grep -q /
++ echo rbac.authorization.k8s.io/v1
++ cut -d/ -f1
+ kcd '-n cattle-fleet-clusters-system Role.rbac.authorization.k8s.io import-token-local-68cdad43-b01a-427f-959d-d0cf2d5797b0-creds'
+ i=0
+ '[' 0 -lt 4 ']'
+ timeout 21 sh -c 'kubectl delete --ignore-not-found=true --grace-period=15 --timeout=20s -n cattle-fleet-clusters-system Role.rbac.authorization.k8s.io import-token-local-68cdad43-b01a-427f-959d-d0cf2d5797b0-creds'
role.rbac.authorization.k8s.io "import-token-local-68cdad43-b01a-427f-959d-d0cf2d5797b0-creds" deleted
+ break
```

</details>

**Comment by superseb** [comment]
@jiaqiluo To confirm a fix, can you share where you ran the script (I guess we only need the bash version).

I guess it comes from https://github.com/rancher/rancher-cleanup/blob/main/cleanup.sh#L68, where it needs to use `"$@"` at the end instead of `"${*}"`.

If this is correct, there would be no issue with cleaning resources as it is used in a log line and the finalizers should be correctly patched. Was there any issue with removing resources or is it just the log line with error?

**Comment by jiaqiluo** [comment]
Hi @superseb,  I ran the cleanup.sh script directly in the command-line terminal.  Resources were removed successfully.

**Comment by superseb** [comment]
@jiaqiluo Thanks, can you share the output of `bash --version` from that terminal?

**Comment by superseb** [comment]
This is in `v1.3.0-rc3`

**Comment by prachidamle** [comment]
@superseb The tag has been updated - this issue is now fixed in `v1.0.4-rc1`
